exercism/python/pangram/pangram.py

```diff
@@ -1,8 +1,73 @@
 import re
+from string import ascii_lowercase as lowercase_letters
 
-
+# set
 def is_pangram(sentence):
-
     return (
         len(set(letter.lower() for letter in re.findall(r"[A-Za-z]", sentence))) == 26
     )
+
+
+# set()
+def is_pangram(sentence):
+
+    alphabet_length = 26
+    sentence_letters = set()
+
+    for letter in sentence.lower():
+        if letter.isalpha():
+            sentence_letters.add(letter)
+
+    return len(sentence_letters) == alphabet_length
+
+
+# dictionary
+def is_pangram(sentence):
+    alphabet = {letter: 0 for letter in lowercase_letters}
+
+    for character in sentence.lower():
+        if character.isalpha():
+            alphabet[character] = alphabet.get(character, 0) + 1
+    for letter, count in alphabet.items():
+        if count == 0:
+            return False
+    return True
+
+
+# Sort and compare
+def is_pangram(sentence):
+
+    sentence_letters = []
+
+    for character in sentence.lower():
+        if character.isalpha() and character not in sentence_letters:
+            sentence_letters.append(character)
+
+    sentence_letters.sort()
+    return sentence_letters == list(lowercase_letters)
+
+
+# array
+def is_pangram(sentence):
+
+    letters = list(lowercase_letters)
+
+    for char in sentence.lower():
+        if char in letters:
+            letter_index = letters.index(char)
+            letters.pop(letter_index)
+    return len(letters) == 0
+
+# array v2
+def is_pangram(sentence):
+    alphabet_length = 26
+    ascii_start = 97
+    total_letters = [0 for i in range(alphabet_length)]
+
+    for i, char in enumerate(sentence):
+        if char.isalpha():
+            #At each letter, get the current ascii val and subtract it from the start of the alphabet 
+            letter_index = ord(char.lower()) - ascii_start 
```

Comment:
`'a'` is identical to `07`, so you can get rid of line 64 and do:

```
ord(char.lower()) - 'a'
```

Reply:
I get errors when I use that:
`letter_index = ord(char.lower()) - 'a'`

but this works:

`letter_index = ord(char.lower()) - ord('a')`

So I added that in and removed the `ascii_start = 97` variable.

Make sense?

Reply:
I like `letter_index = ord(char.lower()) - ord('a')`